extension/lib/updateProductsInCart.js

```diff
@@ -0,0 +1,26 @@
+const ShopgateCartPipeline = require('./shopgate/CartExtensionPipeline')
+
+/**
+ * @param {PipelineContext} context
+ * @param {UpdateProductsInput} input
+ * @returns {Promise<UpdateCartMessage[]>}
+ */
+module.exports = async (context, input) => {
+  try {
+    const warningsPresent = await ShopgateCartPipeline.create(context).updateProducts(input.CartItem)
+    // todo see about proper way to send an warning in case there are non breaking errors present.
+    const messages = []
+    if (warningsPresent) {
+      messages.push({
```

Comment:
@sofib 
This isn't working as we expect it to work. I chatted with Waldemar and he let me know that the only way to deliver a message to the end user would be to add it to the getCart call that happens after the update. The only way I can think to do this would be to write the error to one of the storage elements, and have a get cart to check the storage to see if there is an error add the message to the cart and delete the storage message. Do you think it is worth doing this. Alternatively we could throw an error from the update pipeline. The shopper would see a popup message displaying the error message. After the user clicks okay the check cart runs and all is good. What do you think is the best way to move forward?

Reply:
I'll talk to Waldemar today as well. In general if getCart will be called after the error is thrown we could use this approach and remove everything related to soft error handling.

Reply:
After changing the response of this pipeline to conform to contract messages do show up - tested (they show up as a popup)

Reply:
Thanks for the clarification.